test/integration/controller/jobset/jobset_webhook_test.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jobset
+
+import (
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	testingjob "sigs.k8s.io/kueue/pkg/util/testingjobs/jobset"
+	"sigs.k8s.io/kueue/test/integration/framework"
+	"sigs.k8s.io/kueue/test/util"
+)
+
+var _ = ginkgo.Describe("JobSet Webhook", func() {
+	var ns *corev1.Namespace
+
+	ginkgo.When("with manageJobsWithoutQueueName disabled", ginkgo.Ordered, ginkgo.ContinueOnFailure, func() {
+		ginkgo.BeforeAll(func() {
+			fwk = &framework.Framework{
+				ManagerSetup: managerSetup(),
+				CRDPath:      crdPath,
+				DepCRDPaths:  []string{jobsetCrdPath},
+				WebhookPath:  webhookPath,
+			}
+			ctx, cfg, k8sClient = fwk.Setup()
+		})
+		ginkgo.BeforeEach(func() {
+			ns = &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "jobset-",
+				},
+			}
+			gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+		})
+		ginkgo.AfterEach(func() {
+			gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		})
+		ginkgo.AfterAll(func() {
+			fwk.Teardown()
+		})
+
+		ginkgo.It("the creation doesn't succeed if the queue name is invalid", func() {
+			job := testingjob.MakeJobSet("jobset-with-queue-name", ns.Name).Queue("-").Obj()
```

Comment:
Is this actually rejected by the webhook? I think there might also be an internal validation in k8s for all labels. In particular, when I locally tried to create a Job with a `test` label like this it rejects the job with `The Job "indexed-job" is invalid: metadata.labels: Invalid value: "-": a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')` - even though the label is not covered by Kueue webhook.

I think the test is only valuable if it fails on `main` branch, otherwise it can be misleading.  Is there a way to write the test so that it fails on master?

Reply:
I think `example.com/foo` would pass apiserver validation, but not the webhook.

Reply:
I was trying to use any combinations, but the validation is stopped on internal validation in k8s. In this regard, I have a question, why do we do double check labels?

Reply:
The labels need to satisfy this by k8s
1.: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set
but in our case they are also CRD names, so need to conform to:
2.  https://kubernetes.io/docs/concepts/overview/working-with-objects/names/

So basically, we are looking for a label that passes (1.) but fails (2.). 

The spectrum is narrow, but I think "indexed_job" would do. It passes (1.) as `could contain dashes (-), underscores (_), dots (.), and alphanumerics between.`, but fails (2.) because it contains `_`, which is not enumerated here: `contain only lowercase alphanumeric characters, '-' or '.'`,

Give it a go.